### PR TITLE
module provenance of Wavefunction

### PR DESCRIPTION
**Pull request by loriab**
## Description
Some info about module-level provenance of a Wfn would be handy.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] add `Wavefunction.module_` member data and export
- [x] add setting to most modules (not psimrcc or fockci)
- [x] incorporate into stdsuite testing

## Questions
- [x] @dgasmith I've given mcscf its own name. prefer "detci"?
- [x] @maxscheurer I edited a couple error messages to ADCC from ADC. ok?

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by maxscheurer** [comment]
I would favor "adcc" instead of "ADCC". You agree, @mfherbst?

**Comment by dgasmith** [comment]
I think best to keep it `detci` since the `mcscf` bit is agnostic to the underlying CI code.

File: psi4/src/psi4/dfocc/dfocc.cc [hunk 1]
```diff
@@ -51,6 +51,8 @@ void DFOCC::common_init() {
     print_ = options_.get_int("PRINT");
     if (print_ > 0) options_.print();
 
+    module_ = "dfocc";
```
Comment (dgasmith) [inline psi4/src/psi4/dfocc/dfocc.cc:54]: Just to double check:
- Inherits from Wavefunciton uses `module_ = ...`
- Does not inherit from Wavefunction uses `->set_module()`?

Comment (loriab) [reply psi4/src/psi4/dfocc/dfocc.cc:54]: or, `.set_module()` py-side, yes. I used whatever was closest to the class. No strong opinions, though.

**Comment by mfherbst** [comment]
I agree "adcc" is better ;). Thanks for the cleanup @loriab!

**Comment by loriab** [comment]
> Is SAPT(DFT)/SF-SAPT included with the sapt.cc changes? Since they are mainly python procedures.

I just fixed plain SAPT in the latest commit, as it has to be py-side. I'm not sure what to do about sf-sapt and dft-sapt at this point. Leaving those for now unless other have ideas.

> How are plugins handled? Especially endorsed ones like sns-mp2, v2rdm_cassscf etc.

They'll have to set their own name like https://github.com/piecuch-group/cct3/pull/2/files#diff-dd599d9e2da08b736cbf0b96add9a840R414 or in the pymodule.py file. I've done it for cct3 and adcc (whose py is in psi4/driver) and caspt2 but will have to adjust the other ones you mentioned. And some like pcmsolver, pe, efp/scf won't get recorded. When plugins move to cookiecutter or similar, the module stamping should get baked into the template.

**Comment by loriab** [comment]
> Could go py-sapt (py + procrouting name) perhaps.

sounds reasonable. I'm inclined to leave it blank until needed.

**Pull request merged**